CODE_COMMENT_STYLE.md

```diff
@@ -0,0 +1,116 @@
+# Code Comment Style
+
+This document describes the code comment style applied to TiKV repositories. When you are to commit, be sure to follow the style to write good code comments.
+
+## Why does a good comment matter?
+
+- To speed up the reviewing process
+- To help maintain the code
+- To improve the API document readability
+- To improve the development efficiency of the whole team
+
+## Where/When to comment?
+
+Write a comment where/when there is context that is missing from the code or is hard to deduce from the code. Specifically, use a comment:
+
+- For important code
+- For obscure code
+- For tricky or interesting code
+- For a complex code block
+- If a bug exists in the code but you cannot fix it or you just want to ignore it for the moment
+- If the code is not optimal but you don’t have a smarter way now
+- To remind yourself or others of missing functionality or upcoming requirements not present in the code
+
+A comment is generally required for:
+
+- Module
+- Type
+- Constant
+- Function
+- Method
+- Variable
+- Typical algorithm
+- Exported name
+- Test case
+- TODO
+- FIXME
+
+## How to comment?
+
+### Format of a good comment (for Rust)
+
+- Non-doc comment
+
+    - Used to document implementation details
+    - Use **//** for a line comment
+    - Use **/* … */** for a block comment (used only when needed)
+    
+- Doc comment
+
+    - Used to document the interface of code (structures, fields, macros, etc.)
+    - Use **///** for item documentation (functions, attributes, structures, etc)
+    - Use **//!** for module level documentation
+
+- Place the single-line and block comment above the code it’s annotating
+- Fold long lines of comments
+- The maximum width for a line is 100 characters
+- Use relative URLs when you link to positions on you local machine
+
+### Language for a good comment
+
+- Word
+    
+    - Use **American English** rather than British English
+        
+        - color, canceling, synchronize     (Recommended)
+        - colour, cancelling, synchronise   (Not recommended)
+    
+    - Use correct spelling
+
+    - Use **standard or official capitalization**
+        
+        - TiKV, TiDB-Binlog, Region, gRPC, RocksDB, GC, k8s, [mydumper](https://github.com/maxbube/mydumper), [Prometheus Pushgateway](https://github.com/prometheus/pushgateway)   (Right)
+        - Tikv, TiDB Binlog, region, grpc, rocksdb, gc, k8S, MyDumper, Prometheus PushGateway   (Wrong)
+
+    - Use words and expressions consistently
+        
+        - dead link vs. broken link (Only one of them can appear in a single document)
+    
+    - Do not use lengthy compound words
+
+    - Do not abbreviate unless it is necessary (for readability purposes)
+
+    - "we" should be used only when it means the code writer *and* the reader
+
+- Sentence
+
+    - Use standard grammar and correct punctuation
+    - Use relatively short sentences
+
+- For each comment, capitalize the first letter and end this sentence with a period

```

Comment:
"this sentence" -> "the sentence"

Though I might reword this as "Capitalize the first letter of sentences and end them with periods".

Reply:
I note though that _all_ of the bullet points in this document (which is mostly bullet points) do not end with any punctuation, which to the reader may _appear_ contradictory to this recommendation.